lua/nvim-treesitter/query.lua

```diff
@@ -46,8 +46,20 @@ function M.has_query_files(lang, query_name)
   return files and #files > 0
 end
 
-function M.get_query(lang, query_name)
-  return tsq.get_query(lang, query_name)
+do
+  local cache = {}
+
+  function M.get_query(lang, query_name)
+    if cache[lang] == nil then
+      cache[lang] = {}
+    end
+
+    if cache[lang][query_name] == nil then
+      cache[lang][query_name] = tsq.get_query(lang, query_name)
+    end
+
+    return cache[lang][query_name]
+  end
```

Comment:
when I'm proposed this everyone wanted to have a version that can load the changes from disk. But I'm also for caching the queries as using the queries is a much more frequent use-case than editing and changing the queries

Reply:
I agree because changing the queries is done less often. Also it is slow because if you are running this a lot it runs `io.open` synchronously and then re-parses the whole thing. I talked with bfredl and vigoux on gitter and they suggested that it be cached so thats why i made this pr.

Reply:
Maybe we can have a reload function if people are editing the queries.

Reply:
I think a autocmd for query buffers on write could invalidate the cache

Reply:
I am not sure what you mean?

Reply:
write an autocmd to invalidate cache when you save query buffers

Reply:
`query` is a recognized filetype after installing nvim-treesitter

Reply:
thank you, makes sense now